### PR TITLE
Feat/contribution tree

### PR DESCRIPTION
**Pull request by will7200**
## Checklist

- [x] Create new `NavigatorWidget` class for contribution tree, and make it available in this row:

- [x] Design decision on `tags` system. Tags are a part of [our fundamental data schema](https://github.com/brightway-lca/bw_interface_schemas/blob/5fb1d40587aec2a4bb2248505550fc883a91c355/bw_interface_schemas/lci.py#L68), and are designed to be categorical (quantitative values would normally be a `property`).
- [x] Add ability to add tags to processes, either in the existing edit interface or in a new pop-up. This interface should autocomplete against existing tag systems already in the data; we might want to cache these on the `Database` object instead of scanning the whole database when doing any initial load.
- [x] Implemented aggregation algorithm by `tags` system to the results. Handle missing `tags` elegantly.
- [x] Write initial graph traversal logic in new NavigatorWidget` class 
- [x] Write click event handling logic in `NavigatorWidget` class 
- [ ] User acceptance and change requests

### CODE REVIEW

File: activity_browser/application.py [hunk 1]
```diff
@@ -41,14 +41,14 @@ def deleteLater(self):
...
+    supported = {"10.10", "10.11", "10.12", "10.13", "10.14", "10.15", "13.6"}
     if QSysInfo.productVersion() not in supported:
         os.environ["QT_MAC_WANTS_LAYER"] = "1"
         os.environ["QTWEBENGINE_CHROMIUM_FLAGS"] = "--disable-gpu"
         log.info("Info: GPU hardware acceleration disabled")
 
-if QSysInfo.productType() in ["arch", "nixos"]:
-    os.environ["QTWEBENGINE_CHROMIUM_FLAGS"] = "--no-sandbox"
+if QSysInfo.productType() in ["arch", "nixos", "osx"]:
+    os.environ["QTWEBENGINE_CHROMIUM_FLAGS"] = "{} --no-sandbox".format(os.getenv("QTWEBENGINE_CHROMIUM_FLAGS"))
```
Comment (cmutel) [inline activity_browser/application.py]: This needs a bit of explanation on why and how to set the envvar.

File: activity_browser/ui/web/tree_navigator.py [hunk 2]
```diff
@@ -0,0 +1,602 @@
...
+        return {
+            "nodes": self.nodes,
+            "edges": self.edges,
+            "flows": self.flows,
+            "calculation_count": self.calculation_count.value,
+            "metadata": self.metadata
+        }
+
+    @classmethod
+    def calculate(
```
Comment (cmutel) [inline activity_browser/ui/web/tree_navigator.py]: I'd be a lot happier if we could incorporate missing functionality into `bw_graph_tools`; the AB has done a lot of stuff on it's own in the past, and we now have to fix the messes they created.

**Comment by cmutel** [comment]
Thanks @will7200 .Can we have a call to talk about this? Shouldn't take too long. Maybe in the next 6 hours?

File: activity_browser/application.py (hunk 1 above)
Comment (will7200) [reply activity_browser/application.py]: added the why

File: activity_browser/ui/web/tree_navigator.py (hunk 2 above)
Comment (will7200) [reply activity_browser/ui/web/tree_navigator.py]: moved to bw_graph_tools and cleaned it up a bit as well
https://github.com/brightway-lca/bw_graph_tools/pull/20

**Comment by will7200** [comment]
@cmutel 
For the tag types, since they are aimed at being categorical data types, what data types do we want to support?
The tags support json values so that consist of numbers, strings, arrays, and objects.
Currently I implemented support for: 
+ `int`
+ `float`
+ `string`
+ `date (string underneath)`

So we can drop the number ones (int and float). Do we need to support others ones? How much benefit would supporting lists and objects?

**Comment by cmutel** [comment]
> So we can drop the number ones (int and float). Do we need to support others ones? How much benefit would supporting lists and objects?

A good question - probably the schema is too loose here. I can see a good argument for `int`, in some numbered classification scheme, but am struggling to see how `float` would make sense (especially considering underlying floating point mechanics). I also agree that datetimes smell a bit fishy, but if converted to ISO 8601 then that seems OK.

**Definitely not** objects including lists or any other nonsense. Special cases aren't special enough.

**Comment by will7200** [comment]
@cmutel Ready for review
will require https://github.com/brightway-lca/bw_graph_tools/pull/20 to be merged for tests to pass

**Pull request merged**